File: Arrays/Plus One/python/plus_one.py
def plusOne(digits):
    i = -1
    while i >= -len(digits):
        if digits[i] + 1 < 10:
            digits[i] += 1
            break
        else:
            digits[i] = 0
            i -= 1
    else:
        digits.insert(0, 1)

    return digits


if __name__ == "__main__":
    test_array = [1, 9, 1, 1, 1, 1]
    print(plusOne(test_array))

File: Arrays/Plus One/JS/plus_one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {

  let i = digits.length - 1;

  while (i >= 0){
    if (digits[i] + 1 < 10){
      digits[i]++;
      return digits;
    }
    else {
      digits[i] = 0;
      i--;
    }
  }
  digits.unshift(1)
  return digits;
};
